### PR TITLE
Fix ButtonCascade error in Explore

### PR DESCRIPTION
**Pull request by tskarhed**
Started doing a basic cleanup of the ButtonCascader. While doing that an issue was reported in Explore that children weren't loaded properly. This is beacuse the data format of the cascader alternatives had been changed to `{label: 'label', value:'value', children:'items'}`. The problem was fixed by letting it take the default {label: 'label', value:'value', children:'children'}. Might look into this in the future.

See the `fieldNames` prop.

### CODE REVIEW

File: packages/grafana-ui/src/components/ButtonCascader/ButtonCascader.tsx [hunk 1]
```diff
@@ -8,20 +8,20 @@ import { CascaderOption } from '../Cascader/Cascader';
...
   value?: string[];
+  fieldNames?: any;
 
   loadData?: (selectedOptions: CascaderOption[]) => void;
   onChange?: (value: string[], selectedOptions: CascaderOption[]) => void;
   onPopupVisibleChange?: (visible: boolean) => void;
 }
 
 export const ButtonCascader: React.FC<ButtonCascaderProps> = props => (
-  <RCCascader {...props} fieldNames={{ label: 'label', value: 'value', children: 'items' }}>
```
Comment (dehrax) [inline packages/grafana-ui/src/components/ButtonCascader/ButtonCascader.tsx]: This fixed the error

**Comment by aocenas** [comment]
@dehrax does this also fixes how this look in explore? Or do we need to restyle it? It looked different compared to all the other explore buttons.

File: packages/grafana-ui/src/components/ButtonCascader/ButtonCascader.tsx [hunk 2]
```diff
@@ -8,20 +8,20 @@ import { CascaderOption } from '../Cascader/Cascader';
 
 export interface ButtonCascaderProps {
   options: CascaderOption[];
-  buttonText: string;
+  children: string;
   disabled?: boolean;
-  expandIcon?: React.ReactNode;
   value?: string[];
+  fieldNames?: any;
```
Comment (aocenas) [inline packages/grafana-ui/src/components/ButtonCascader/ButtonCascader.tsx]: Shouldn't be a `{ [key: string]: string }` or something like that?

**Comment by dehrax** [comment]
This one uses the button from the new form styles. Would I go with the regular button instead?

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx [hunk 3]
```diff
@@ -32,6 +32,7 @@ export interface CascaderOption {
   items?: CascaderOption[];
   disabled?: boolean;
   title?: string;
+  [key: string]: any;
```
Comment (aocenas) [inline packages/grafana-ui/src/components/Cascader/Cascader.tsx]: Can we constrain this type a bit or can it really contain any key?

**Comment by aocenas** [comment]
@dehrax 
> This one uses the button from the new form styles. Would I go with the regular button instead?

Honestly no idea. I mean right now it looks weird so we should fix it in some way just not sure what is the most sensible one. Maybe add an optional button class so that you can inject different button element? cc @dprokop

**Comment by ivanahuckova** [comment]
Not related to Explore, but it also looks a little bit off in dashboards, as the button is very different than the rest. I really like how the button looks, but I think we should change then labels, fields and buttons as well, or keep it consistent. 
image

**Comment by dehrax** [comment]
Alternative:

Screenshot from 2020-01-17 17-00-27
Screenshot from 2020-01-17 17-00-40

We can probably wait with the new button until we start using the new form components in all the other places.

**Comment by dehrax** [comment]
Compared with original @ivanahuckova 
Screenshot from 2020-01-17 17-03-24

**Comment by ivanahuckova** [comment]
Yeah, I agree with what you've said - to wait with the new button until we start using the new form components in all the other places - and use the original button for now. 🙂

**Review by ivanahuckova — APPROVED** [review]
@aocenas had 2 good points/questions related to typing. If you could look at that, it would be awesome! Other than that, everything looks and works good.

**Pull request merged**